tpp-app/src/home/components/Selector.js

```diff
@@ -45,31 +45,51 @@ const Selector = (props) => {
+        },
+        {
+            onPress: () => props.toggleSelectedView(VIEWS.Cramps),
+            touchableStyle: [crampsSelected && styles.selectedIcon, styles.iconContainer],
+            internalIcon: (<CrampsIcon style={styles.icon} fill= {crampsSelected ? selectedColor : unselectedColor}/>)
+        },
+        {
+            onPress: () => props.toggleSelectedView(VIEWS.Sleep),
+            touchableStyle: [sleepSelected && styles.selectedIcon, styles.iconContainer],
+            internalIcon: (<SleepIcon style={styles.icon} fill= {sleepSelected ? selectedColor : unselectedColor}/>)
+        },
+    ]
+
+
 
     return (
         <View style={[{backgroundColor: '#FFFFFF'}, props.expanded && styles.elevatedSelector]}>
             {props.expanded &&
             <View style={[styles.selectorContainer]}>
-                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Flow)} 
-                    style={[flowSelected && styles.selectedIcon, styles.iconContainer]} >
-                    <FlowIcon style={styles.icon} fill={flowSelected ? selectedColor : unselectedColor}/>
-                </TouchableOpacity>
-                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Mood)} 
-                    style={[ moodSelected && styles.selectedIcon, styles.iconContainer]}>
-                    <MoodIcon style={styles.icon} fill= {moodSelected ? selectedColor : unselectedColor}/>
-                </TouchableOpacity>
-                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Exercise)} 
-                    style={[exerciseSelected && styles.selectedIcon, styles.iconContainer]}>
-                    <ExerciseIcon style={styles.icon} fill= {exerciseSelected ? selectedColor : unselectedColor}/>
-                </TouchableOpacity>
-                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Cramps)} 
-                    style={[ crampsSelected && styles.selectedIcon, styles.iconContainer]}>
-                    <CrampsIcon style={styles.icon} fill= {crampsSelected ? selectedColor : unselectedColor}/>
-                </TouchableOpacity>
-                <TouchableOpacity onPress={() => props.toggleSelectedView(VIEWS.Sleep)} 
-                    style={[sleepSelected && styles.selectedIcon, styles.iconContainer]}>
-                    <SleepIcon style={styles.icon} fill= {sleepSelected ? selectedColor : unselectedColor}/>
-                </TouchableOpacity>
+                 {iconData.map((icon, i) => {
+                    return (
+                        <TouchableOpacity
+                            onPress={icon.onPress}
+                            key={i}
+                            style={icon.touchableStyle}
+                        >
+                            {icon.internalIcon}
+                        </TouchableOpacity>
+                    )
+                })}
```

Comment:
```suggestion
                 {iconData.map((icon, i) => {
                    let renderedIcon = createElement(icon.internalIcon, {
                        fill: icon.selected ? selectedColor : unselectedColor
                    });
                    return (
                        <TouchableOpacity
                            onPress={() => props.toggleSelectedView(icon.view)} // not 100% positive this will work, like 90%
                            key={i}
                            style={[icon.selected && styles.selectedIcon, styles.iconContainer]}
                        >
                            {renderedIcon}
                        </TouchableOpacity>
                    )
                })}
```
Make sure you import `createElement` from React.